src/App.tsx

```diff
@@ -1,49 +1,63 @@
-import { Provider } from 'react-redux'
-import { RouterProvider, createBrowserRouter } from 'react-router-dom'
-import { AuthLayout, MasterLayout, NotFound } from './Components'
-import { AddNewRoom, Ads, Bookings, Facilities, ForgetPassword, Home, Login, Register, ResetPassword, Rooms, Users } from './Pages'
-import Store from './Redux/Store'
-
-
+import { Provider } from "react-redux";
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import { AuthLayout, MasterLayout, NotFound } from "./Components";
+import {
+  AddNewRoom,
+  Ads,
+  Bookings,
+  Facilities,
+  ForgetPassword,
+  Home,
+  Login,
+  Register,
+  ResetPassword,
+  Rooms,
+  Users,
+} from "./Pages";
+import Store from "./Redux/Store";
+import ProtectedRoute from "./Components/ProtectedRoute/ProtectedRoute";
 
 function App() {
-
   const routes = createBrowserRouter([
-
     {
-      path: "dashboard", element: <MasterLayout />, errorElement: <NotFound />, children: [
+      path: "dashboard",
+      element: (
+        <ProtectedRoute>
+          <MasterLayout />
+        </ProtectedRoute>
+      ),
+      errorElement: <NotFound />,
+      children: [
         { index: true, element: <Home /> },
-        { path: 'users', element: <Users /> },
-        { path: 'rooms', element: <Rooms /> },
-        { path: 'add-new-room', element: <AddNewRoom /> },
-        { path: 'facilities', element: <Facilities /> },
-        { path: 'ads', element: <Ads /> },
-        { path: 'bookings', element: <Bookings /> },
-      ]
+        { path: "users", element: <Users /> },
+        { path: "rooms", element: <Rooms /> },
+        { path: "add-new-room", element: <AddNewRoom /> },
```

Comment:
For better page hierarchy, you can change this path to `rooms/add-new-room` to indicate that this page is reached from the rooms page.